sklearn/decomposition/fastica_.py

```diff
@@ -348,9 +352,19 @@ def g(x, fun_args):
     del X1
 
     if whiten:
-        if compute_sources:
-            S = np.dot(np.dot(W, K), X).T
-        else:
+        # if compute_sources:
+        #     S = np.dot(np.dot(W, K), X).T
+        #     S_std = np.std(S, axis=0, keepdims=True)
+        #     S /= S_std
+        #     W /= S_std.T
+        # else:
+        #     S = None
+
+        S = np.dot(np.dot(W, K), X).T
```

Comment:
there's no need to compute `S` if `compute_sources` is `False`, this would be unnecessary computation.

Reply:
I know, but if I don't do that, the value of `W` is wrong (by a factor of 10 in the test). I couldn't find a way to update `W` without computing `S`. Any ideas?

Reply:
Fair point! I guess in that case it's also fair to update the description of `compute_sources` parameter, maybe.

Reply:
Great! If that's okay, I'll delete the commented lines (`355` to `361`).

Regarding the parameter `compute_sources` would it be something like this?

```
compute_sources: bool, optional
    Sources are computed in any case because they are needed to compute the 
    estimated un-mixing matrix. If False, sources are not returned (default=True).
```